### PR TITLE
DOC-456 | Create JavaScript section and re-organize JS API content

### PR DESCRIPTION
**Pull request by Simran-B**
Also disperse the Appendix

Applied to 3.10 and 3.11 as well

Needs example re-generation.

### CODE REVIEW

File: 3.12/javascript-api.md [hunk 1]
```diff
@@ -104,64 +288,3 @@ The following Node.js modules are not available at all:
 - `tls`
```
Comment (nerpaula) [inline 3.12/javascript-api.md:288]: Let's add links for:
- but see `@arangodb/crypto` below
- but see `@arangodb/request` below

File: 3.12/appendix-java-script-modules-arango-db.md [hunk 2]
```diff
@@ -19,10 +19,9 @@ compatible with each other and cannot be used interchangeably.
 The `db` object
 ---------------
 
-`arangodb.db`
-
 The `db` object represents the current database and lets you access collections
-and run queries. For more information see the [db object reference](appendix-references-dbobject.html).
+and Views, run AQL queries, and more. For more information, see
+[The `db` object](appendix-references-dbobject.html).
```
Comment (nerpaula) [inline 3.12/appendix-java-script-modules-arango-db.md:24]:
```suggestion
the [`db` object](appendix-references-dbobject.html).
```

**Review by nerpaula — COMMENTED** [review]
LGTM. Let's apply the changes to 3.10 and 3.11 as well.

**Review by nerpaula — APPROVED** [review]

**Comment by Simran-B** [comment]
Started Jenkins against v3.10.6 and v3.11.0 tags, as well as devel branch:
https://jenkins.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/3225/

**Pull request merged**